### DISCUSSION TITLE
Problem when compare PANDAR networks

### DISCUSSION THREAD

**Discussion by thangnx1012** [netZoo/netZooR#291] [Q&A] [answered]
Hello,
I am interested in how PANDA work to create networks that fits for my study. However, I got an error when comparing data between two PANDA networks using the PandaR package, following this tutorial (https://rdrr.io/github/netZoo/netZooR/f/vignettes/pandaRApplicationinGTExData.Rmd) or even my dataset.

(everything work well until)

Plot the network
visDiffPandaInCytoscape(diffRes_vis, condition_name = "High_Agressive", network_name="diff.PANDA")
Error in visDiffPandaInCytoscape(diffRes_vis, condition_name = "High_Agressive", :
could not find function "visDiffPandaInCytoscape"
Apply the style to the network. Purple edges indicate higher edge weight in the defined "condition_name" parameter (HOS in our example), and green edges indicate higher edge weight in the other condition (MG63 in our example).
createDiffPandaStyle(style_name="Diff.PandaStyle", condition_name="High_Agressive")
Error in createDiffPandaStyle(style_name = "Diff.PandaStyle", condition_name = "High_Agressive") :
could not find function "createDiffPandaStyle"

I tried to copy the visDiffPandaInCytoscape function (https://rdrr.io/github/netZoo/netZooR/src/R/visDiffPandaInCytoscape.R) and createDiffPandaStyle function (https://rdrr.io/github/netZoo/netZooR/src/R/createDiffPandaStyle.R) into R, visDiffPandaInCytoscape run well so createDiffPandaStyle did not work. The console showed the message like this.

Plot the network
visDiffPandaInCytoscape(diffRes_vis, condition_name = "High_Agressive", network_name="diff.PANDA")
You are connected to Cytoscape!
Loading data...
Applying default style...
Applying preferred layout...
networkSUID
38404
Apply the style to the network. Purple edges indicate higher edge weight in the defined "condition_name" parameter (HOS in our example), and green edges indicate higher edge weight in the other condition (MG63 in our example).
createDiffPandaStyle(style_name="Diff.PandaStyle", condition_name="High_Agressive")
Error in createDiffPandaStyle(style_name = "Diff.PandaStyle", condition_name = "High_Agressive") :
unused arguments (style_name = "Diff.PandaStyle", condition_name = "High_Agressive")

Please help me out with this issue.

sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4.1
Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib; LAPACK version 3.11.0
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
time zone: Europe/Oslo
tzcode source: internal
attached base packages:
[1] stats4 stats graphics grDevices utils datasets methods base
other attached packages:
[1] devtools_2.4.5 usethis_2.2.2 tibble_3.2.1 RCy3_2.20.1 visNetwork_2.1.2 viridisLite_0.4.2 limma_3.56.2 reshape2_1.4.4
[9] ggplot2_3.4.3 fgsea_1.26.0 dplyr_1.1.2 topGO_2.52.0 SparseM_1.81 GO.db_3.17.0 AnnotationDbi_1.62.2 IRanges_2.34.1
[17] S4Vectors_0.38.1 graph_1.78.0 openxlsx_4.2.5.2 data.table_1.14.8 netZooR_1.4.0 matrixcalc_1.0-6 yarn_1.26.0 pandaR_1.32.0
[25] Biobase_2.60.0 BiocGenerics_0.46.0 reticulate_1.31 igraph_1.5.1
loaded via a namespace (and not attached):

**Comment by marouenbg** [comment]
Hi @thangnx1012 ,
Thanks for letting me know, you can find an updated vignette here: https://github.com/netZoo/netZooR/pull/294

**Reply by thangnx1012** [reply]
Hi Marouenbg,
Thank you for the reply. I do see the updated version on netbooks, so I'm not quite satisfied with the visNetwork plot as messy and untouchable plot. You can see what it looks like with my plot in the attachment. Could you guide me on how to manage the plot with Cytoscape?
Best regards
Thang

https://github.com/netZoo/netZooR/assets/49030920/579a9a02-ec19-4add-b378-2025dc53774d

**Reply by marouenbg** [reply]
Hi @thangnx1012 ,
We've been using visNetwork for our publication figures. For cytoscape, you need to export the network as csv file, but I don't have an example to share with you but from this function https://rdrr.io/github/netZoo/netZooR/src/R/visDiffPandaInCytoscape.R it seems that you need to export the netwrok as a dataframe with "source","target","motif","weight" columns.

**Reply by thangnx1012** [reply]
Hi Marouenbg,
As I mentioned in the first question, I will copy it here for more clarity:

"I tried to copy the visDiffPandaInCytoscape function (https://rdrr.io/github/netZoo/netZooR/src/R/visDiffPandaInCytoscape.R) and createDiffPandaStyle function (https://rdrr.io/github/netZoo/netZooR/src/R/createDiffPandaStyle.R) into R source, visDiffPandaInCytoscape run well so createDiffPandaStyle did not work. The console showed the message like this.

Plot the network
visDiffPandaInCytoscape(diffRes_vis, condition_name = "High_Agressive", network_name="diff.PANDA")
You are connected to Cytoscape!
Loading data...
Applying default style...
Applying preferred layout...
networkSUID
38404

#Apply the style to the network. Purple edges indicate higher edge weight in the defined "condition_name" parameter (HOS in our example), and green edges indicate higher edge weight in the other condition (MG63 in our example).

createDiffPandaStyle(style_name="Diff.PandaStyle", condition_name="High_Agressive")
Error in createDiffPandaStyle(style_name = "Diff.PandaStyle", condition_name = "High_Agressive") :
unused arguments (style_name = "Diff.PandaStyle", condition_name = "High_Agressive")"

I have no idea how to modify this createDiffPandaStyle function. Do you have any suggestions?

Best regards,
Thang

**Reply by marouenbg** [reply, marked as answer]
I see now. I did that too and it worked for me!
<img width="1165" alt="Screen Shot 2023-09-02 at 7 56 05 
<img width="684" alt="Screen Shot 2023-09-02 at 7 56 11 PM" src="https://github.com/netZoo/netZooR/assets/7914548/0d70e287-f78c-492f-8960-7be627adc892">
PM" src="https://github.com/netZoo/netZooR/assets/7914548/93f3646d-22a9-4d0a-a962-49091fb75095">
Try copying this function instead https://github.com/netZoo/netZooR/blob/master/R/createDiffPandaStyle.R

**Comment by thangnx1012** [comment]
Hi Marouenbg,

It's finally worked for me now. Thank you for saving me and being so supportive.

Best wishes,
Thang

**Reply by malonzm1** [reply]
Hi. Why are the edges not colored purple or green?
